### PR TITLE
`clickButton` can submit forms

### LINKED ISSUE #80: Simulate submitting a form
having that html view function

```elm
viewForm form =
    Html.form [ onSubmit SubmittedForm ]
        [ fieldset [ class "form-group" ]
            [ input
                [ class "form-control"
                , placeholder "username"
                , onInput EnteredEmail
                , value form.email
                , autofocus True
                , type_ "email"
                ]
                []
            ]
        , fieldset [ class "form-group" ]
            [ input
                [ class "form-control"
                , type_ "password"
                , placeholder "Password"
                , onInput EnteredPassword
                , value form.password
                ]
                []
            ]
        , button [ class "btn btn-lg btn-primary" ]
            [ text "Sign in" ]
        ]
```

and trying to test this through:
```elm
   test "Login Form - shows errors when submitting empty username" <|
            \_ ->
                ProgramTest.createElement
                    { ... }
                    |> ProgramTest.start ()
                    |> clickButton "Sign in"
                    |> expectViewHas [ Selector.text "User id can't be blank." ]
        ]
```

causes the following error message:
```
✗ Login Form - shows errors when submitting empty username

    clickButton "Sign in":
    Event.expectEvent: I found a node, but it does not listen for "click" events like I expected it would.
    
    <button class="btn btn-lg btn-primary">
        Sign in
    </button>
```

How could I test that form (I don't want to remove the `onSubmit` in favour of `onClick` on the `button` element unless that is the only way to do it).

### PR DESCRIPTION
**Pull request by avh4**
Fixes #80 

Probably easier to review commit-by-commit

### CODE REVIEW

File: tests/ProgramTestTests/UserInput/ClickButtonTest.elm [hunk 1]
```diff
@@ -55,11 +55,12 @@ all =
...
+                        , "▼ Query.has [ text \"HTML expected by the call to: clickButton \"Click Me\"\" ]"
                         , ""
-                        , "0 matches found for this query."
+                        , "✗ has text \"HTML expected by the call to: clickButton \"Click Me\"\""
                         , ""
-                        , ""
-                        , "✗ Query.find always expects to find 1 element, but it found 0 instead."
+                        , "Expected one of the following to exist:"
+                        , "- <button> (not disabled) with text \"Click Me\""
+                        , "- an element with role=\"button\" (not disabled) and text \"Click Me\""
```
Comment (mthadley) [inline tests/ProgramTestTests/UserInput/ClickButtonTest.elm]: Love how this prints out all of the checks now. 🎉

File: src/ProgramTest.elm [hunk 2]
```diff
@@ -880,15 +886,35 @@ clickButton buttonText programTest =
...
+                                    , Selector.attribute (Html.Attributes.value buttonText)
+                                    ]
+                              ]
+                            ]
+                        , onError =
+                            [ Selector.tag "form"
+                            , Selector.containing
+                                [ Selector.tag "input"
+                                , Selector.attribute (Html.Attributes.type_ "submit")
+                                , Selector.attribute (Html.Attributes.disabled False)
```
Comment (mthadley) [inline src/ProgramTest.elm]: Nice job guarding against disabled buttons! Too bad there isn't a way to include this check in the base/good query.

File: src/ProgramTest.elm [hunk 3]
```diff
@@ -822,19 +822,35 @@ If the button is disabled the test will fail.
...
-            , findNotDisabled
-                [ Selector.attribute (Html.Attributes.attribute "role" "button")
-                , Selector.containing [ Selector.text buttonText ]
-                ]
+    let
+        functionDescription =
+            "clickButton " ++ escapeString buttonText
+
+        checks =
+            [ ( "<button> (not disabled) with text " ++ escapeString buttonText
```
Comment (avh4) [inline src/ProgramTest.elm]: Also add that the button doesn't have an onclick handler

Comment (avh4) [reply src/ProgramTest.elm]: ^^ (from Katie)

**Pull request merged**